js/models/messages.js

```diff
@@ -136,6 +142,17 @@
 
             return '';
         },
+        /* eslint-enable */
+        /* jshint ignore:start */
+        async onDestroy() {
+          this.revokeImageUrl();
+          const attachments = this.get('attachments');
+          const deleteData =
+            Attachment.deleteData(migrationContext.deleteAttachmentData);
```

Comment:
In terms of API, I would like to have the final `deleteData` available to me here as an end-user, instead of needing to assemble it with `context`. I can see why we arrange things this way at declaration, keeping things pure, keeping things easy to test and compose. But I think we could simplify things here by having things be pre-assembled.

Reply:
Agreed, that’d be nice. I will have to think about how to best expose it since the regular upgrade pipeline just needs the `writeAttachmentData` function on its own.